skycoin/skywire
# Add stun NAT type check for sudph

Did you run `make format && make check`?
yes

Fixes #776 

 Changes:	
- Added a `STUN` check for the `sudph` transport.
- Separated transports into their own modules so that they can be initialized individually and the visor will start only when at least one transport has been initialized. 
- `dmsg` and `sudph` initialize separate from other modules so that the visor doesn't have to wait for them. 
- Removed the unused `public` field from AddTransports.

How to test this PR:
1. Run `./skywire-visor skywire-config.json`
2. If your visor is under a symmetric NAT you should not be able to see a `sudph` transport. (check the logs to see your NAT type)
